projects/forge-angular/src/lib/dialog/dialog.service.ts

```diff
@@ -92,4 +103,41 @@ export class DialogService {
     dialogInstance.open = false;
     ref.destroy();
   }
+
+  public ngOnDestroy(): void {
+    this._unsubscribe$.next(null);
+    this._unsubscribe$.complete();
+  }
+
+  /**
+   * Closes all dialogs.
+   * @param result The result of closing the dialogs. Default is false.
+   */
+  public closeAllDialogs(result = false): void {
+    this._closeAllDialogs(result);
+  }
+
+  private _closeAllDialogs(result: boolean, recursiveExecutionCount = 0): void {
+    if (recursiveExecutionCount > 2) {
```

Comment:
@DRiFTy17 This is needed for the nested dialogs.  I have it set to 2.  is there a reasonable number that it should be set to?  I know within our app we do not allow dialogs to open dialogs just as UX functionality, but I know our app does not represent the full usage of forge.

Reply:
So this is just a safe guard against it recursing infinitely basically in the case where closing a dialog also opens a dialog? This is probably an edge case, but I understand the need for it. I would say `2` sounds fine to me, but let's put that into a constant like `MAX_NESTED_DIALOGS` to avoid the magic number. If we ever run into a case where this needs to be adjusted then we could make the value configurable or something, but this looks fine to me for now.

Reply:
Done